libs/sdk-core/src/swap_in/swap.rs

```diff
@@ -194,384 +234,283 @@ impl BTCReceiveSwap {
 
-        let swap_info = SwapInfo {
-            bitcoin_address: swap_reply.bitcoin_address,
-            created_at: SystemTime::now()
-                .duration_since(UNIX_EPOCH)
-                .unwrap()
-                .as_secs() as i64,
-            lock_height: swap_reply.lock_height,
-            payment_hash: hash.clone(),
-            preimage: swap_keys.preimage,
-            private_key: swap_keys.priv_key.to_vec(),
-            public_key: pubkey.clone(),
-            swapper_public_key: swap_reply.swapper_pubkey.clone(),
-            script: our_script.as_bytes().to_vec(),
-            bolt11: None,
-            paid_msat: 0,
-            unconfirmed_sats: 0,
-            confirmed_sats: 0,
-            total_incoming_txs: 0,
-            refund_tx_ids: Vec::new(),
-            confirmed_tx_ids: Vec::new(),
-            unconfirmed_tx_ids: Vec::new(),
-            status: SwapStatus::Initial,
-            min_allowed_deposit: swap_reply.swapper_min_payable,
-            max_allowed_deposit: fn_max_allowed_deposit(swap_reply.swapper_max_payable),
-            max_swapper_payable: swap_reply.swapper_max_payable,
-            last_redeem_error: None,
-            channel_opening_fees: Some(channel_opening_fees),
-            confirmed_at: None,
-        };
-        swap_info.validate_swap_limits()?;
-
-        // persist the swap info
-        self.persister.insert_swap(swap_info.clone())?;
+        let swap_info = self
+            .taproot
+            .create_swap(&node_state, opening_fee_params)
+            .await?;
+        self.persister.insert_swap(&swap_info)?;
         Ok(swap_info)
     }
 
-    fn list_unused(&self) -> Result<Vec<SwapInfo>> {
-        Ok(self.persister.list_swaps(ListSwapsRequest {
-            status: Some(SwapStatus::unused()),
-            ..Default::default()
-        })?)
+    pub(crate) fn list_swaps(&self, req: ListSwapsRequest) -> ReceiveSwapResult<Vec<SwapInfo>> {
+        Ok(self.persister.list_swaps(req)?)
     }
 
-    pub(crate) fn list_in_progress(&self) -> Result<Vec<SwapInfo>> {
-        Ok(self.persister.list_swaps(ListSwapsRequest {
+    pub(crate) fn list_in_progress_swaps(&self) -> ReceiveSwapResult<Vec<SwapInfo>> {
+        self.list_swaps(ListSwapsRequest {
             status: Some(SwapStatus::in_progress()),
             ..Default::default()
-        })?)
+        })
     }
 
-    pub fn list_monitored(&self) -> Result<Vec<SwapInfo>> {
-        Ok(self.persister.list_swaps(ListSwapsRequest {
+    pub fn list_monitored(&self) -> ReceiveSwapResult<Vec<SwapInfo>> {
+        let monitored = self.list_swaps(ListSwapsRequest {
             status: Some(SwapStatus::monitored()),
             ..Default::default()
-        })?)
-    }
-
-    pub(crate) fn list_refundables(&self) -> Result<Vec<SwapInfo>> {
-        Ok(self.persister.list_swaps(ListSwapsRequest {
-            status: Some(SwapStatus::refundable()),
+        })?;
+        let recent = self.list_swaps(ListSwapsRequest {
```

Comment:
So we are rescanning anyway all swaps until timeout? regardless of their status?  Just curious what have we been missing before this implementation.

Reply:
This is just to make sure we catch additional funds sent to the same address within a reasonable time.